### PR TITLE
Removed validator, auto increment for roundID

### PR DESCRIPTION
**Pull request by KelvinThai**
# Description

This PR remove validator in aggregator, set auto increment for RoundID

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [x] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: contracts-v0.2/src/Aggregator.sol [hunk 1]
```diff
@@ -128,20 +110,17 @@ contract Aggregator is Ownable, IAggregator, ITypeAndVersion {
...
-        oracleInitializeNewRound(uint32(_roundId));
-        recordSubmission(_submission, uint32(_roundId));
-        (bool updated, int256 newAnswer) = updateRoundAnswer(uint32(_roundId));
-        deleteRoundDetails(uint32(_roundId));
-        if (updated) {
-            validateAnswer(uint32(_roundId), newAnswer);
-        }
+        oracleInitializeNewRound(uint32(roundId));
+        recordSubmission(_submission, uint32(roundId));
+        updateRoundAnswer(uint32(roundId));
```
Comment (martinkersner) [inline contracts-v0.2/src/Aggregator.sol:122]: The return values from `updateRoundAnswer` are not used anymore. Shouldn't we update the function as well?

File: contracts-v0.2/src/Aggregator.sol [hunk 2]
```diff
@@ -128,20 +110,17 @@ contract Aggregator is Ownable, IAggregator, ITypeAndVersion {
...
     /**
      * @notice called by oracles when they have witnessed a need to update
-     * @param _roundId is the ID of the round this submission pertains to
      * @param _submission is the updated data that the oracle is submitting
      */
-    function submit(uint256 _roundId, int256 _submission) external {
-        bytes memory error = validateOracleRound(msg.sender, uint32(_roundId));
+    function submit(int256 _submission) external {
+        uint32 roundId = reportingRoundId + 1;
+        bytes memory error = validateOracleRound(msg.sender, uint32(roundId));
```
Comment (martinkersner) [inline contracts-v0.2/src/Aggregator.sol:117]: There is no need to convert `roundId` to `uint32` anywhere in this function (every line of this function).

File: contracts-v0.2/src/Aggregator.sol [hunk 3]
```diff
@@ -173,12 +152,7 @@ contract Aggregator is Ownable, IAggregator, ITypeAndVersion {
             addOracle(_added[i]);
         }
 
-        updateFutureRounds(
-            _minSubmissionCount,
-            _maxSubmissionCount,
-            _restartDelay,
-            timeout
-        );
+        updateFutureRounds(_minSubmissionCount, _maxSubmissionCount, _restartDelay, timeout);
```
Comment (martinkersner) [inline contracts-v0.2/src/Aggregator.sol:155]: We should not modify the the code that is not meant to be changed. Please roll back all the formatting changes. We should not mix formatting PRs with functionality changes.

File: contracts-v0.2/src/Aggregator.sol [hunk 4]
```diff
@@ -412,30 +375,11 @@ contract Aggregator is Ownable, IAggregator, ITypeAndVersion {
...
      */
-    function typeAndVersion()
-        external
-        pure
-        virtual
-        override
-        returns (string memory)
-    {
+    function typeAndVersion() external pure virtual override returns (string memory) {
         return "Aggregator v0.1";
```
Comment (martinkersner) [inline contracts-v0.2/src/Aggregator.sol:383]: Wrong version of `Aggregator`.

**Review by martinkersner — CHANGES_REQUESTED** [review]

File: contracts-v0.2/test/Aggregator.t.sol [hunk 5]
```diff
@@ -81,28 +80,12 @@ contract AggregatorTest is Test {
...
-        aggregator.submit(1, 11);
-
-        vm.prank(address(2));
-        vm.expectEmit(true, true, false, true);
-        emit AnswerUpdated(11, 1, timestamp);
-        aggregator.submit(1, 12);
-
-        (, int256 answer, , , ) = aggregator.latestRoundData();
-        assertEq(answer, 11);
+        aggregator.submit(10);
```
Comment (nick-bisonai) [inline contracts-v0.2/test/Aggregator.t.sol:88]: I think it's missing read after submit.

File: contracts-v0.2/test/Aggregator.t.sol [hunk 6]
```diff
@@ -179,39 +162,16 @@ contract AggregatorTest is Test {
...
             vm.warp(timestamp + i);
             vm.prank(address(0));
-            aggregator.submit(i, 321);
+            aggregator.submit(321);
             uint256 startedAt = aggregator.currentRoundStartedAt();
             assertEq(startedAt, timestamp + i);
         }
     }
 
     function test_validateOracleRound() public {
```
Comment (nick-bisonai) [inline contracts-v0.2/test/Aggregator.t.sol:171]: Checking if roundId is actually auto incremented could be included

File: contracts-v0.2/src/Aggregator.sol (hunk 1 above)
Comment (KelvinThai) [reply contracts-v0.2/src/Aggregator.sol:122]: hmm! I see. Can we fully remove this function, then update `latestRoundID` inside of `Submit` function?

Comment (martinkersner) [reply contracts-v0.2/src/Aggregator.sol:122]: Not now, you would have to make many other updates as well.

**Review by nick-bisonai — APPROVED** [review]
lgtm!

**Review by martinkersner — APPROVED** [review]
LGTM!

**Pull request merged**